Add French translation

lbourdois opened the pull request:
Hi @MKhalusova, 


In this PR you'll find the French translation of the course.
Only chapter 8 is missing. This is deliberate, as the deadline for certification is September 1, and the French translation will probably not be finished by then, so translating content that will no longer be viable when it's released doesn't seem relevant to me.

Indeed, most of the work is done, but in my experience, markdown tags or typos (spelling, keystrokes, etc.) have probably pop in. These are generally not obvious in the `mdx` files and require the `doc-builder`, which strangely doesn't work locally on my side, or which is accessible once have `write access`, which I don't yet have on this repo. 
Correcting them takes a few days and is fairly easy, but I'll be off from tomorrow evening for several weeks, so they probably won't arrive before September 1, although my aim is to be able to offer this complete translation by early September for the start of the university year.


Have a nice day,

Loïck

MKhalusova commented on chapters/fr/chapter5/asr_models.mdx:
````diff
@@ -0,0 +1,245 @@
+```python
+print(sample["text"])
+Audio(sample["audio"]["array"], rate=sample["audio"]["sampling_rate"])
+```**Sortie :**
````
The formatting here results in doc-builder errors.  A new line before `**Sortie :**` should fix it.

lbourdois replied:
Done

MKhalusova commented on chapters/fr/chapter5/asr_models.mdx:
````diff
@@ -0,0 +1,245 @@
+C'est le texte cible que nous visons avec notre transcription Whisper. Bien que nous sachions maintenant que nous pouvons probablement faire mieux, puisque notre modèle va aussi prédire la ponctuation et la casse, qui ne sont pas présents dans la référence. Transmettons l'échantillon audio au pipeline pour obtenir notre prédiction de texte. Une chose à noter est que le pipeline _consomme_ le dict ...
+```python
+pipe(sample["audio"].copy(), max_new_tokens=256, generate_kwargs={"task": "transcribe"})
+```**Sortie :**
````
Same formatting issue as above.

lbourdois replied:
Done

MKhalusova commented on chapters/fr/chapter1/audio_data.mdx:
```diff
@@ -0,0 +1,203 @@
+Étant donné que les séquences sont différentes pour les exemples audio à des taux d'échantillonnage différents, il sera difficile pour les modèles de généraliser entre les taux d'échantillonnage.
+**Le rééchantillonnage** est le processus de mise en correspondance des taux d'échantillonnage et fait partie du [prétraitement](preprocessing#resampling-the-audio-data) des données audio.
+
+## Amplitude et profondeur de bits
```
Please add a new line after the headers.

MKhalusova reviewed (changes requested):
Thank you for taking over this massive effort and translating the whole course! 
Two more files need to be updated in order to build the course in a new language. `.github/workflows/build_documentation.yml` and `.github/workflows/build_pr_documentation.yml`.
Please add `fr` to the list of languages similar to how it's done [here](https://github.com/huggingface/audio-transformers-course/blob/360ba6cffda44d249029f790f03a93b08ec74cb8/.github/workflows/build_documentation.yml#L17).

Also, please review the markdown formatting. I spotted a couple of places where it breaks - see comments.

lbourdois commented:
> Two more files need to be updated in order to build the course in a new language. .github/workflows/build_documentation.yml and .github/workflows/build_pr_documentation.yml.

Ah well, now I understand why I wasn't able to run the doc builder on my side. 🙃

I've corrected what you pointed out.

MKhalusova commented:
The checks are failing ATM. Please run `make style` locally and try to build locally to identify the issues.

lbourdois commented:
Classic NLP course mistake, shame on me for not thinking to do this line of code before opening my PR.
It should be fine now :)

MKhalusova commented:
Almost there! The `audio_data.mdx` is missing a header, and that breaks the build.

lbourdois commented:
I hope it will be good by now. If it isn't, I won't be able to make any changes until 3 weeks from now when I get back from vacation.

MKhalusova commented:
Unfortunately, there's still an issue with building the docs. This time it appears to be formatting in `chapter3/quiz.mdx:135:174`. 
I would recommend to try building locally to catch this and other possible issues.

lbourdois commented:
Having had quick access to the internet while on vacation, I tried to look at the problem quickly.
But not having my computer with me, I did it on my phone and on the browser.
After doing some crap, I simply deleted the quiz for the moment.
If it goes well, great, if not, I'll do a cleaner job after my vacation.

lbourdois commented:
Hi @MKhalusova 
Indeed the `force-pushed main branch from 4d14d38 to ab97af9` replaced the `fr` with a `ru`. I re-added it.

MKhalusova commented on chapters/fr/chapter6/tts_datasets.mdx:
```diff
@@ -0,0 +1,40 @@
+[LJSpeech](https://huggingface.co/datasets/lj_speech) est un jeu de données composé de 13 100 échantillons audio de langue anglaise jumelés à leurs transcriptions correspondantes. Le jeu de données contient l'enregistrement d'un seul locuteur lisant des phrases de 7 livres en anglais. LJSpeech est souvent utilisé comme référence pour évaluer les modèles de TTS en raison de sa haute qualité audio  ...
+
+## Multilingual LibriSpeech
+[Multilingual LibriSpeech] (https://huggingface.co/datasets/facebook/multilingual_librispeech) est une extension multilingue du jeu de données LibriSpeech, qui est à la base une collection à grande échelle de livres audio lus en anglais. LibriSpeech multilingue inclus des langues supplémentaires, telles que l'allemand, le néerlandais, l'espagnol, le français, l'italien, le portugais et le polonai ...
```
```suggestion
[Multilingual LibriSpeech](https://huggingface.co/datasets/facebook/multilingual_librispeech) est une extension multilingue du jeu de données LibriSpeech, qui est à la base une collection à grande échelle de livres audio lus en anglais. LibriSpeech multilingue inclus des langues supplémentaires, telles que l'allemand, le néerlandais, l'espagnol, le français, l'italien, le portugais et le polonais. 
```

MKhalusova commented on chapters/fr/chapter6/tts_datasets.mdx:
```diff
@@ -0,0 +1,40 @@
+
+[LJSpeech](https://huggingface.co/datasets/lj_speech) est un jeu de données composé de 13 100 échantillons audio de langue anglaise jumelés à leurs transcriptions correspondantes. Le jeu de données contient l'enregistrement d'un seul locuteur lisant des phrases de 7 livres en anglais. LJSpeech est souvent utilisé comme référence pour évaluer les modèles de TTS en raison de sa haute qualité audio  ...
+
+## Multilingual LibriSpeech
```
```suggestion
## Multilingual LibriSpeech

```

MKhalusova commented on chapters/fr/chapter6/tts_datasets.mdx:
```diff
@@ -0,0 +1,40 @@
+Il contient des enregistrements audio de 110 anglophones avec divers accents. Chaque orateur lit environ 400 phrases, qui ont été choisies dans un journal, et un paragraphe d'élicitation utilisé pour les archives de l'accent de la parole.
+C’est une ressource précieuse pour l’entraînement de modèles de TTS avec des voix et des accents variés, permettant une synthèse vocale plus naturelle et diversifiée.
+
+## Libri-TTS/ LibriTTS-R
```
```suggestion
## Libri-TTS/ LibriTTS-R

```

MKhalusova commented on chapters/fr/chapter6/pre-trained_models.mdx:
````diff
@@ -0,0 +1,199 @@
+Audio(speech, rate=16000)
+```
+
+Superbe ! Si vous souhaitez essayer MMS avec une autre langue, trouvez d'autres *checkpoints* `vits` appropriés sur le [*Hub*] (https://huggingface.co/models?filter=vits).
````
```suggestion
Superbe ! Si vous souhaitez essayer MMS avec une autre langue, trouvez d'autres *checkpoints* `vits` appropriés sur le [*Hub*](https://huggingface.co/models?filter=vits).
```

MKhalusova commented on chapters/fr/chapter7/voice-assistant.mdx:
```diff
@@ -0,0 +1,361 @@
+
+## Détection des mots déclencheur
+
+La première étape du pipeline de l'assistant vocal consiste à détecter si le mot déclencheur a été prononcé, et nous devons donc trouver un modèle pré-entraîné approprié pour cette tâche ! Vous vous souviendrez de la section sur les [modèles pré-entraînés pour la classification audio](../chapiter4/classification_models) que [Speech Commands](https://huggingface.co/datasets/speech_commands) est un ...
```
```suggestion
La première étape du pipeline de l'assistant vocal consiste à détecter si le mot déclencheur a été prononcé, et nous devons donc trouver un modèle pré-entraîné approprié pour cette tâche ! Vous vous souviendrez de la section sur les [modèles pré-entraînés pour la classification audio](../chapiter4/classification_models) que [Speech Commands](https://huggingface.co/datasets/speech_commands) est un jeu de données de mots prononcés conçu pour évaluer les modèles de classification audio sur plus de 15 mots de commande simples comme `up`, `down`, `yes` et `no`, ainsi qu'un label `silence` pour classer l'absence de parole. Prenez une minute pour écouter les échantillons sur le visualisateur de jeux de données du *Hub* et vous familiariser à nouveau avec le jeu de données : [visualisateur](https://huggingface.co/datasets/speech_commands/viewer/v0.01/train).
```

MKhalusova commented on chapters/fr/chapter7/voice-assistant.mdx:
```diff
@@ -0,0 +1,361 @@
+
+## Synthétiser la parole
+
+Et maintenant, nous sommes prêts à obtenir la sortie vocale finale ! Une fois de plus, nous utiliserons le modèle [SpeechT5 TTS] (https://huggingface.co/microsoft/speecht5_tts) de Microsoft mais vous pouvez utiliser n'importe quel modèle TTS de votre choix. Chargeons le processeur et le modèle :
```
```suggestion
Et maintenant, nous sommes prêts à obtenir la sortie vocale finale ! Une fois de plus, nous utiliserons le modèle [SpeechT5 TTS](https://huggingface.co/microsoft/speecht5_tts) de Microsoft mais vous pouvez utiliser n'importe quel modèle TTS de votre choix. Chargeons le processeur et le modèle :
```

MKhalusova commented on chapters/fr/chapter7/supplemental_reading.mdx:
```diff
@@ -0,0 +1,18 @@
+Le matériel de lecture supplémentaire est donc divisé en ces trois nouvelles tâches :
+
+Traduction vocale (audio à audio) :
+* [STST avec unités discrètes] (https://ai.facebook.com/blog/advancing-direct-speech-to-speech-modeling-with-discrete-units/) par Meta AI : une approche directe de la STST par le biais de modèles encodeur-décodeur.
```
```suggestion
* [STST avec unités discrètes](https://ai.facebook.com/blog/advancing-direct-speech-to-speech-modeling-with-discrete-units/) par Meta AI : une approche directe de la STST par le biais de modèles encodeur-décodeur.
```

MKhalusova reviewed (changes requested):
We’re very close to merging this epic PR! Now that the pages are building, I can see that there are multiple places where there are issues with formatting, specifically with rendering links and code examples.

To fix these, please add:
- a new line before and after a code block 
- a new line after each header 
- a new line before and after a `<Tip>…</Tip>` block, a `<div>…</div>` block

I have also left a few formatting nits in the comments. Thanks for your work on this PR!

lbourdois commented:
I've added a new line before and after a code block, after each header, before and after a <Tip>…</Tip> block,  and before a <div>…</div> block. And I've corrected the broken links.
I hope everything is ok.

lbourdois commented:
Fingers crossed it's finally good

MKhalusova reviewed (approved):
Great work! 
We can merge now. Unit 8 (congratulations and how to get a certificate) is missing, but feel free to add it in a separate PR.
